### PR TITLE
[JSDoc] Add missing documentation for botbuilder-dialogs-adaptive recognizers folder's root

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 2602

## Description
This PR adds all the missing JSDoc documentation for [botbuilder-dialogs-adaptive](https://github.com/microsoft/botbuilder-js/tree/main/libraries/botbuilder-dialogs-adaptive) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

## Specific Changes

- Ported documentation from botbuilder-dotnet
- Added documentation for the files inside `recognizers` folder's root.
- Added `@private` for private methods.
- Added `@protected` for protected methods.
- Added JSDoc comments in **6 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive/src/recognizers/intentPattern.ts [hunk 1]
```diff
@@ -7,39 +7,70 @@
...
         this._pattern = value.startsWith('(?i)') ? value.substr(4) : value;
     }
 }
 
+/**
+ * Input pattern converter that implements `Converter`.
+ */
 export class IntentPatternConverter implements Converter {
+    /**
+     * Converts an object or string to a `IntentPattern` instance.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/recognizers/intentPattern.ts:70]:
```suggestion
     * Converts an object or string to an `IntentPattern` instance.
```

File: libraries/botbuilder-dialogs-adaptive/src/recognizers/recognizerSet.ts [hunk 2]
```diff
@@ -10,10 +10,21 @@ import { RecognizerResult, Activity, getTopScoringIntent } from 'botbuilder-core
 import { DialogContext } from 'botbuilder-dialogs';
 import { Recognizer } from './recognizer';
 
+/**
+ * RecognizerSet - Recognizer which is the union of of multiple recognizers into one RecognizerResult.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/recognizers/recognizerSet.ts:14]:
```suggestion
 * RecognizerSet - Recognizer which is the union of multiple recognizers into one RecognizerResult.
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Comment by sw-joelmut** [comment]
@ceciliaavila feedback applied, thanks

**Review by iscaltritti-sw — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
promoted

**Pull request closed without merging**